src/layouts/sidebar-sidecar/server.ts

```diff
@@ -123,6 +127,45 @@ export function getStaticGenerationFunctions<
         basePath,
       ])
 
+      /**
+       * constructs the levels of nav data for DocsView pages
+       *
+       * @TODO - do something with it?
+       */
+      const sidebarPropsLevels = [
```

Comment:
This will be what a lot of views will have to do.

Reply:
It feels like we could abstract this up another level, to something like `navData`, as ideally I think we'll look to pull from this data to power the Hashi logo dropdown menu.

With that in mind, it might make sense to add an `id` property to allow easier consumption, what do you think?

I'm thinking we make the navData available in a context, and allow for selection. In the Hashi menu I would for example do this:

```ts
useNavDataForLevel('main-menu')
```

(also acknowledging I might be thinking a bit too far ahead here, but figured it'd be worth discussing)

Reply:
I really like the idea of this so far. I think it's too early to add to this PR, but let's continue discussing in Slack! It might make getting this working in tutorial-view and collection-view a little easier. 🤔

Reply:
Added the new `Context`, resolving this thread! ✨